### PR TITLE
Rasterization State data group

### PR DESCRIPTION
**Pull request by hliatis**
Includes data from Rasterization bits and code to handle bitfields

### CODE REVIEW

**Comment by hliatis** [comment]
Bug: 141623574

File: cmd/gapit/dump_pipeline.go [hunk 1]
```diff
@@ -110,6 +111,17 @@ func toString(dataval *api.DataValue) string {
...
+
+	case *api.DataValue_Bitfield:
+		var b strings.Builder
+		for i, bitname := range x.Bitfield.SetBitnames {
+			if i == len(x.Bitfield.SetBitnames)-1 {
+				fmt.Fprintf(&b, "%s", bitname)
+			} else {
+				fmt.Fprintf(&b, "%s | ", bitname)
+			}
+		}
```
Comment (pmuetschard) [inline cmd/gapit/dump_pipeline.go]: Can you use `strings.Join` here?

File: gapic/src/main/com/google/gapid/views/PipelineView.java [hunk 2]
```diff
@@ -338,6 +356,20 @@ private String convertToString(API.DataValue val) {
...
+
+        for (int i = 0; i < val.getBitfield().getSetBitnamesCount(); i++) {
+          builder.append(val.getBitfield().getSetBitnames(i));
+
+          if (i < val.getBitfield().getSetBitnamesCount() - 1) {
+            builder.append(" | ");
+          }
+        }
+
+        return builder.toString();
```
Comment (pmuetschard) [inline gapic/src/main/com/google/gapid/views/PipelineView.java]: Use the guava `Joiner.on(" | ")...`

File: gapis/api/vulkan/resources.go [hunk 3]
```diff
@@ -1197,9 +1197,23 @@ func (p GraphicsPipelineObjectʳ) geometryShader() *api.Stage {
 }
 
 func (p GraphicsPipelineObjectʳ) rasterizer() *api.Stage {
+	rasterState := p.RasterizationState()
+	rasterList := &api.KeyValuePairList{}
+	rasterList = rasterList.AppendKeyValuePair("Depth Clamp Enabled", api.CreatePoDDataValue("VkBool32", rasterState.DepthClampEnable() != 0))
+	rasterList = rasterList.AppendKeyValuePair("Rasterizer Discard", api.CreatePoDDataValue("VkBool32", rasterState.RasterizerDiscardEnable() != 0))
+	rasterList = rasterList.AppendKeyValuePair("Polygon Mode", api.CreateEnumDataValue("VkPolygonMode", rasterState.PolygonMode()))
+	rasterList = rasterList.AppendKeyValuePair("Cull Mode", api.CreateBitfieldDataValue("VkCullModeFlags", rasterState.CullMode(), VkCullModeFlagBitsConstants(), api.Find(api.ID(ID))))
```
Comment (pmuetschard) [inline gapis/api/vulkan/resources.go]: `api.Find(api.ID(ID))` is the same as `API{}` (see the generated `api.go` file for each API).

Comment (hliatis) [reply gapis/api/vulkan/resources.go]: Done. I would still need to pass it to the function no matter what, correct?

File: gapic/src/main/com/google/gapid/views/PipelineView.java [hunk 4]
```diff
@@ -239,18 +244,32 @@ protected void onUiThread(API.Pipeline result) {
...
 
-                    withLayoutData(createLabel(dataComposite, convertToString(kvp.getValue())),
+                    withLayoutData(createLabel(kvpComposite, convertToString(kvp.getValue())),
                       new GridData(SWT.BEGINNING, SWT.TOP, false, false));
                   }
 
+                  scrollComposite.setContent(contentComposite);
+                  scrollComposite.setExpandVertical(true);
+                  scrollComposite.setExpandHorizontal(true);
+                  scrollComposite.addListener( SWT.Resize, event -> {
```
Comment (pmuetschard) [inline gapic/src/main/com/google/gapid/views/PipelineView.java]: nit: extra space.

**Pull request merged**